packages/bpk-component-scrollable-calendar/src/BpkScrollableCalendarGridList.js

```diff
@@ -49,71 +51,74 @@ class BpkScrollableCalendarGridList extends React.Component {
       startDate,
     );
     const months = getMonthsArray(startDate, monthsCount);
+    // Here we calculate the height of each calendar grid item in pixels, as the `react-window` API
+    // requires that these are provided so that they can be efficiently rendered.
+    const monthItemHeights = [];
+    for (let i = 0; i < months.length; i += 1) {
+      const firstDayOffset = (months[i].getDay() + 7 - props.weekStartsOn) % 7;
+      const monthLength = DateUtils.daysInMonth(
+        months[i].getYear(),
+        months[i].getMonth(),
+      );
+      const calendarGridSpaces = firstDayOffset + monthLength;
+      const rowCount = Math.ceil(calendarGridSpaces / COLUMN_COUNT);
+      monthItemHeights[i] = GRID_HEIGHT_WITHOUT_ROWS + ROW_HEIGHT * rowCount;
+    }
 
-    const cache = new CellMeasurerCache({
-      fixedWidth: true,
-      defaultHeight: 276, // most common height (in px) of BpkScrollableCalendarGrid
-    });
     this.state = {
       months,
-      cache,
+      monthItemHeights,
+      outerHeight: ESTIMATED_MONTH_ITEM_HEIGHT,
     };
   }
 
+  componentDidMount = () => {
+    this.setComponentHeight();
+    if (typeof window !== 'undefined') {
+      document.addEventListener('resize', this.setComponentHeight);
+      document.addEventListener('orientationchange', this.setComponentHeight);
+      document.addEventListener('fullscreenchange', this.setComponentHeight);
```

Comment:
It feels weird that we check `window` is defined but then add event listeners to `document`. Should these listeners be added to the window? Or should the check be that `document` is defined?